ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/RequiredPluginsClasspathContainer.java

```diff
@@ -61,13 +63,18 @@
 public class RequiredPluginsClasspathContainer extends PDEClasspathContainer implements IClasspathContainer {
 
 	@SuppressWarnings("nls")
-	private static final Collection<String> JUNIT5_RUNTIME_PLUGINS = Set.of("org.junit", "junit-jupiter-api",
-			"junit-jupiter-engine", "junit-platform-commons", "junit-platform-engine", "org.hamcrest.core",
-			"org.opentest4j");
+	private static final Set<String> JUNIT5_RUNTIME_PLUGINS = Set.of("org.junit", //
```

Comment:
I wonder why `org.junit` (i.e. the Junit-4 bundle is added), if a Test-Plugin references the JUnit-5 API?
Can anyone tell?

Reply:
You can use JUnit 4 API in JUnit5 tests via the `junit-vintage-engine` (even though it is not really a JUnit5 test then) so one is able to mix JUnit 3/4/5 ...

Reply:
Yes, but then one imports the org.junit packge or requires that bundle?
Or is the intention to make it available without any further addo once one imports the Jupiter-API?

Reply:
Actually require it to importpackage/requirebundle junit is bad, why should I ever want that? Beside that you should look at the JUnit Container at JDT that defines the set of junit jars and I think PDE is inspired by that.

Actually one wants to define the used engine (like in BND Tests or Surefire) and then it might be enough to determine transitive requirements.

Reply:
> Actually require it to importpackage/requirebundle junit is bad, why should I ever want that?

I more think, why I should ever not want that? All other dependencies needed for compilation have to be specified, why should a Junit dependency used for compilation (not runtime), be different in this case?
We have only dedicated 'test-plugins/fragments' in PDE, so there is IMHO no great need to treat them differently.
And as said in another comment, I have the impression that this can lead to compilation failures in Tycho because in the IDE the classpath sees more jars than in the build.